lib/xcodeproj/plist_helper.rb

```diff
@@ -63,6 +64,37 @@ def read(path)
 
       private
 
+      # Simple workaround to escape characters which are outside of ASCII
+      # character-encoding. Relies on the fact that there are no XML characters
+      # which would need to be escaped.
+      #
+      # @note   This is necessary because Xcode (4.6 currently) uses the MacRoman
+      #         encoding unless the `// !$*UTF8*$!` magic comment is present. It
+      #         is not possible to serialize a plist using the NeXTSTEP format
+      #         without access to the private classes of Xcode and that comment
+      #         is not compatible with the XML format. For the complete
+      #         discussion see CocoaPods/CocoaPods#926.
+      #
+      #
+      # @note   Sadly this hack is not sufficient for supporting Emoji.
+      #
+      # @param  [String, Pathname] The path of the file which needs to be fixed.
+      #
+      # @return [void]
+      #
+      def fix_encoding(filename)
+        output = ''
+        input = File.open(filename, 'rb') { |file| file.read }
+        input.unpack('U*').each do |codepoint|
+          if codepoint > 127
```

Comment:
Perhaps might be worth mentioning that the ASCII range is 0-127 in a comment so it's clear what this magic number is about.